static/js/src/app/users.js

```diff
@@ -115,6 +115,35 @@ const deleteUser = (id) => {
     })
 }
 
+const swapUser = (id) => {
+    var user = users.find(x => x.id == id)
+    if (!user) {
+        return
+    }
+    Swal.fire({
+        title: "Are you sure?",
+        html: "You will be logged out of your account and logged in as <strong>" + escapeHtml(user.username) + "</strong>",
+        type: "warning",
+        animation: false,
+        showCancelButton: true,
+        confirmButtonText: "Swap User",
+        confirmButtonColor: "#428bca",
+        reverseButtons: true,
+        allowOutsideClick: false,
+    }).then((result) => {
+        if (result.value) {
+          fetch('/superlogin', {
```

Comment:
Perhaps a different approach would be to have an HTML form that points to `/impersonate` in the table itself, and the modal just submits that form? I think we have a similar pattern elsewhere, but I could be wrong. Just something to consider, that way we're not having to encode the CSRF token like this 🤷 But if that's not feasible, then approach will work fine.

Reply:
So I could use a form by putting this HTML in the users.html:

```
<!-- Form for Impersonate-->
<form id="form_imp" style="visibility: hidden" action="/impersonate" method="POST">
    <input name="username" type="text" value="monzo">
    <input id="csrf_imp" name="csrf_token" type="text">
    <input type="submit" value="Submit">
</form> 
```

and then in the users.js:

```
$("#user_imp").val(user.username)
$("#csrf_imp").val(csrf_token)
$("#form_imp").submit()
```

The problem with this would fall back to the above issue of figuring out how to display a message; if we perform a submit we'll load a whole new page before getting a change to pop up a dialogue, so I think we'll need to stick with the javascript approach.